jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8325,13 +8325,157 @@ public String indent(int n) {
 /*[ENDIF] Java12 */
 
 /*[IF Java13]*/
-	/*
+	/**
 	 * Determine if current String object is LATIN1.
 	 *
 	 * @return true if it is LATIN1, otherwise false.
 	 */
 	boolean isLatin1() {
 		return LATIN1 == coder();
 	}
+
+	/**
+	 * Format the string using this string as the format with supplied args.
+	 *
+	 * @param args
+	 *          the format arguments to use
+	 *
+	 * @return the formatted result
+	 *
+	 * @see #format(String, Object...)
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String formatted(Object... args) {
+		return String.format(this, args);
+	}
+
+	/**
+	 * Removes the minimum indentation from the beginning of each line and
+	 * removes the trailing spaces in every line from the string
+	 *
+	 * @return this string with incidental whitespaces removed from every line
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String stripIndent() {
+		if (isEmpty()) {
+			return "";
+		}
+		Iterator<String> iter = lines().iterator();
+		StringBuilder builder = new StringBuilder();
+		int min = Integer.MAX_VALUE;
+
+		while (iter.hasNext()) {
+			String line = iter.next();
+
+			if ((!line.isBlank()) || (!iter.hasNext())) {
+				int count = 0;
+				int limit = Math.min(min, line.length());
+				while ((count < limit) && (Character.isWhitespace(line.charAt(count)))) {
+					count++;
+				}
+				min = (count < min ? count : min);
+			}
+
+			if (line.isBlank()) {
+				builder.append("");
+			} else {
+				builder.append(line.stripTrailing());
+			}
+			if (iter.hasNext()) {
+				builder.append("\n");
+			}
+		}
+
+		String result = builder.toString().indent(-min);
+		if (charAt(length() - 1) != '\n' && charAt(length() - 1) != '\r') {
```

Comment:
This will fail if the result is empty.

Reply:
I have added a empty check at the beginning of the method

Reply:
Please extract `length()` into a local variable: it may be used three times here.